public/components/datasources/components/manage/accelerations/create_accelerations_flyout/create/create_acceleration_button.tsx

```diff
@@ -39,7 +40,7 @@ export const CreateAccelerationButton = ({
 
     const requestPayload: DirectQueryRequest = {
       lang: 'sql',
-      query: accelerationQueryBuilder(accelerationFormData),
+      query: accelerationQueryBuilder(accelerationFormData).replaceAll(SANITIZE_QUERY_REGEX, ' '),
```

Comment:
I know I have [code that's subject to the same issue](https://github.com/opensearch-project/dashboards-observability/blob/76d206c18bf412d9c4c541f06c991843a113ac7d/public/components/integrations/components/setup_integration.tsx#L446), but this will also replace instances of multiple spaces in query strings and prevents things like adding single-line comments to queries.

It might be beyond the scope of this PR, but if we're doing this repeatedly, I think we should make a centralized `sanitizeSql` method that can handle detecting spaces in strings and remove comments before it tries to normalize whitespace.

Reply:
Agreed. We need also centralize direct query calls so that we  keep `sanitizeSql` method used only in one location. Have added this hook that can be used at all places from cache loaders, integrations to https://github.com/opensearch-project/dashboards-observability/blob/76d206c18bf412d9c4c541f06c991843a113ac7d/public/framework/datasources/direct_query_hook.tsx

Issue: https://github.com/opensearch-project/dashboards-observability/issues/1503

Reply:
added comments to the issue we are tracking